traits/observation/tests/test_has_traits_helpers.py

```diff
@@ -137,159 +136,146 @@ def test_iter_objects_does_not_trigger_property(self):
         self.assertEqual(foo.property_n_calculations, 0)
 
 
+class CannotCompare:
+
+    def __eq__(self, other):
+        raise TypeError("Cannot be compared for equality.")
+
+
 class ObjectWithEqualityComparisonMode(HasTraits):
     """ Class for supporting TestHasTraitsHelpersWarning """
 
     list_values = List(comparison_mode=2)
     dict_values = Dict(comparison_mode=2)
     set_values = Set(comparison_mode=2)
-    property_list = Property(List(comparison_mode=2))
-    container_in_union = Union(
-        None,
-        Set(comparison_mode=1),
-        comparison_mode=2,
-    )
+    number = Any(comparison_mode=2)
```

Comment:
Off-topic for this PR, but we should change these to use the `ComparisonMode` constants.

Reply:
Open #1229 for this, but I don't want to cause merge conflicts for this PR. So we should update #1229 after merging this PR.

Reply:
OK!  I don't mind resolving the merge conflicts here either. Either way is fine for me.